api-module-library/hubspot/tests/api.test.js

```diff
@@ -299,36 +299,50 @@ describe(`${config.label} API tests`, () => {
             primaryLandingPages = await api.getLandingPages('translatedFromId__is_null');
             expect(primaryLandingPages).toBeDefined();
         });
-        let variationLandingPages
+        let variationLandingPages;
+        let sampleLandingPage;
         it('should return only variation language landing pages', async () => {
             variationLandingPages = await api.getLandingPages('translatedFromId__not_null');
             expect(variationLandingPages).toBeDefined();
+            sampleLandingPage = variationLandingPages.results.slice(-1)[0];
+            expect(sampleLandingPage.id).toBeDefined();
         });
         it('confirm total landing pages', async () => {
             expect(allLandingPages.total).toBe(primaryLandingPages.total + variationLandingPages.total)
         });
+
         it('get Landing Page by Id' , async () => {
-            const pageToGet = variationLandingPages.results.slice(-1)[0];
-            const response = await api.getLandingPage(pageToGet.id);
+            const response = await api.getLandingPage(sampleLandingPage.id);
             expect(response).toBeDefined();
         });
-        it('update a Landing page (minimal patch)' , async () => {
-            const pageToUpdate = variationLandingPages.results.slice(-1)[0];
+        it('update a Landing page (maximal patch)' , async () => {
+            delete sampleLandingPage['archivedAt'];
             const response = await api.updateLandingPage(
-                pageToUpdate.id,
-                {htmlTitle: `test Landing page ${Date.now()}`},
+                sampleLandingPage.id,
+                sampleLandingPage,
                 true );
             expect(response).toBeDefined();
         });
-        it('update a Landing page (maximal patch)' , async () => {
-            const pageToUpdate = variationLandingPages.results.slice(-1)[0];
-            delete pageToUpdate['archivedAt'];
+        it('update a Landing page (minimal patch)' , async () => {
             const response = await api.updateLandingPage(
-                pageToUpdate.id,
-                pageToUpdate,
+                sampleLandingPage.id,
+                {htmlTitle: `test Landing page ${Date.now()}`},
                 true );
             expect(response).toBeDefined();
         });
+        it('publish a Landing Page' , async () => {
+            const now = new Date(Date.now()+5000);
+            const response = await api.publishLandingPage(
+                sampleLandingPage.id,
+                now.toISOString(),
+            );
+            expect(response).toBeDefined();
+        });
+        it('push a Landing page draft to live' , async () => {
+
+            const response = await api.pushLandingPageDraftToLive(sampleLandingPage.id);
+            expect(response).toBeDefined();
```

Comment:
Is it possible to check a few more details (e.g., HTTP response code, body details expected via `expect(response).toStrictEqual(expect.objectContaining(...))`?

Giving more clues about the expected response here would help to make the target excerpt more resilient to future refactoring.

Reply:
There's an underlying issue with the Requester class that the API inherits from that returns an empty string for successful post with no body in the response. I could modify the options to get the full response object, and check the status on that. I've done that previously, but I also don't like exposing the underlying node implementation (i.e. passing back the fetch response object).

I could also change the test to `expect(response).toBe("");` but that seems like going in the wrong direction / makes the test brittle to actually fixing the post handling in Requester. 

Because anything other than an underlying 200 status response will cause this to fail, I'd like to just keep it as is.

Reply:
This specific request returns an empty response, is what I'm taking from this @MichaelRyanWebber. Is that accurate?

Side note, for future requests if we ever need/want the raw response, there's a helper boolean option of `returnFullRes` https://github.com/friggframework/frigg/blob/96d612cbe023fdf1694ec9ec1913296e125b1e57/packages/module-plugin/requester/requester.js#LL77C25-L77C26

That's mostly for more complex requests that involve things like grabbing Location headers for redirects or working with binary etc.

Reply:
@seanspeaks yes it's an empty response. For posts the Requester turns that into an empty string rather than a response object with an empty body but with a status and headers. Same with patch and put. I consider it to be a bug, but I'm not %100.

And yeah, that's what i was talking about in my comment, I could switch these requests to `returnFullRes: true` but to me that obfuscates things, since that full response is a node-fetch response that uses getters for some things, whereas Requester returns an object. 

So rather than encoding current behavior (by testing for "") or working around it by having each request like this grab the fullResponse, pull status from it etc, just leave as is with the thought to improve Requester's handling of this.

Reply:
@MichaelRyanWebber - as any non-200 will cause an error (I assume it would raise a rejected promise, right?), we could rewrite this test like this then:

```
const response = api.pushLandingPageDraftToLive(sampleLandingPage.id);

// If the processing fails, Jest will say "Received promise rejected instead of resolved" instead of some Javascript-based error that is usually harder to track down
await expect(response).resolves.toBe(/* any value you may be expecting to come from the output */);
```

It would give a more meaningful output in case of failure since you don't have a tangible output to check against.

Reply:
https://linear.app/lefthook/issue/LEF-441/re-asses-requesters-handling-of-empty-response-body-on-postpatchput

to continue conversation